vnext/Microsoft.ReactNative/Utils/TextHitTestUtils.cpp

```diff
@@ -0,0 +1,327 @@
+  const auto maxLeft = 0;
+  const auto maxRight = static_cast<float>(start.VisualParent().Width());
+
+  // Check if run is either the start or end of a line
+  const auto characterBeforeStart = GetBoundaryCharacter(start, startRect, -1);
+  const auto characterAfterEnd = GetBoundaryCharacter(end, endRect, 1);
+
+  // Find leftmost character of the first line and first LTR character
+  BidirectionalTextBoundary startBoundary;
+  startBoundary.ltrPointer = start;
+  startBoundary.ltrRect = startRect;
+  if (isStartRTL) {
+    startBoundary = FindStartRTLBoundary(start, end, startRect);
+  }
+
+  // Find rightmost character of the last line and last LTR character
+  BidirectionalTextBoundary endBoundary;
+  endBoundary.ltrPointer = end;
+  endBoundary.ltrRect = endRect;
+  if (isEndRTL) {
+    endBoundary = FindEndRTLBoundary(start, end, endRect);
+  }
+
+  // Step 1: The lines between the first and last lines
+  const auto middleTop = startBottom;
+  const auto middleBottom = endRect.Y;
+  if (middleTop <= targetPoint.Y && middleBottom >= targetPoint.Y) {
+    return true;
+  }
+
+  // Step 2: The LTR characters on the first line before initial RTL
+  // Optimization b: Skip if no LTR characters in first line
+  if (startBoundary.ltrPointer) {
+    auto left = startBoundary.ltrRect.X;
+    if (!characterBeforeStart.has_value()) {
+      left = maxLeft;
+    }
+
+    auto right = maxRight;
+    if (isSingleLine && characterAfterEnd.has_value()) {
+      // TODO: last LTR character is ignored because width is always 0
+      right = endBoundary.ltrRect.X + endBoundary.ltrRect.Width;
+    }
+
+    if (IsInBounds(targetPoint, left, right, startTop, startBottom)) {
+      return true;
+    }
+  }
+
+  // Step 3: The LTR characters on the last line after final RTL
+  // Optimization b: Skip if no LTR characters in last line
+  if (!isSingleLine && endBoundary.ltrPointer) {
+    const auto left = 0;
+    // TODO: last LTR character is ignored because width is always 0
+    auto right = endBoundary.ltrRect.X + endBoundary.ltrRect.Width;
+    if (!characterAfterEnd.has_value()) {
+      right = maxRight;
+    }
+
+    if (IsInBounds(targetPoint, left, right, endTop, endBottom)) {
+      return true;
+    }
+  }
+
+  // Step 4: The initial RTL characters on the first line
+  // Optimization c: Skip if first line starts with LTR
+  if (isStartRTL) {
+    // TODO: last RTL character is ignored because width is always 0
+    auto left = startBoundary.rtlRect.X + startBoundary.rtlRect.Width;
+    if (!characterBeforeStart.has_value()) {
+      left = maxLeft;
+    }
+
+    auto right = maxRight;
+    if (isSingleLine) {
+      right = startRect.X;
+    }
+
+    if (IsInBounds(targetPoint, left, right, startTop, startBottom)) {
+      return true;
+    }
+  }
+
+  // Step 5: The final RTL characters on the last line
+  // Optimization d: Skip if last line ends with LTR
+  if (isEndRTL) {
+    // TODO: last RTL character is ignored because width is always 0
```

Comment:
is this TODO resolved by now?

Reply:
Unfortunately it is not. I'll file an issue and reference it here.

Reply:
Can you reference these new issues in the PR description once you create them?